index.html

```diff
@@ -2395,39 +2403,106 @@
+          </tr>
+        </table>
       </p>
+
       <p class="note">
-        Though <code>'*'</code> matches a sequence of characters, only URLs
-        with scheme <code>"https"</code> are accepted in the algorithms
-        taking a <a>URL pattern</a>. The <code>'*'</code> is a valid
-        character for the URL path component, therefore,
-        <code>'https://www.mydomain.com/*'</code> pattern will match
+        For example, <code>'https://mydomain.com/*'</code> will match
+        <code>'https://service.mydomain.com/myapp/'</code> and
+        <code>'https://info.mydomain.com/general/'</code>, while
+        <code>'https://app.mydomain.com/contacts'</code> will match
+        <code>'https://app.mydomain.com/contacts'</code> and
+        <code>'https://app.mydomain.com/contacts/all'</code>
+
+        The <code>'*'</code> is a valid character for the URL path component,
+        therefore,
+        <code>'https://www.mydomain.com/*'</code> pattern will match both
         <code>'https://www.mydomain.com/*'</code> and
         <code>'https://www.mydomain.com/service'</code> URLs.
       </p>
-      <p>
-        For example, <code>'*://*.mydomain.com/*'</code> will match
-        <code>'https://services.mydomain.com/myservice1/myapp2/'</code> and
-        <code>'https://info.mydomain.com/general/'</code>.
-      </p>
     </section>
+
+    <section> <h3>URL pattern match algorithm</h3>
+        To <dfn>match Web NFC Id with URL pattern</dfn> for a given
+        <code><a>Web NFC Id</a></code> and <code><a>URL pattern</a></code>, run
+        these steps:
+        <ol>
+          <li>
+            Let <var>id</var> be a <code><a>Web NFC Id</a></code> passed to this algorithm.
+          </li>
+          <li>
+            Let <var>pattern</var> be a <a>URL pattern</a> passed to this algorithm.
+          </li>
+          <li>
+            If <var>id</var> and <var>pattern</var> are empty <code>String</code>s,
+            return <code>true</code> and abort these steps.
+          </li>
+          <li>
+            Let <code>id_url</code> be the result of invoking the
+            <a>basic URL parser</a> algorithm with <var>id</var> as an input.
+            If the <a>basic URL parser</a> algorithm returns
+            <code>failure</code>, return <code>false</code> and abort these steps.
+          </li>
+          <li>
+            Let <code>pattern_url</code> be the result of invoking the
+            <a>basic URL parser</a> algorithm with <var>pattern</var> as an input.
+            If the <a>basic URL parser</a> algorithm returns
+            <code>failure</code>, return <code>false</code> and abort these steps.
+          </li>
+          <li>
+            If <code>id_url</code>'s <a href="#dfn-url-scheme">scheme</a> <code>String</code>
+            does not match <code>pattern_url</code>'s <a href="#dfn-url-scheme">scheme</a>
+            <code>String</code>, return <code>false</code> and abort these steps.
+          </li>
```

Comment:
I think this is good general formulation: elsewhere the spec restricts applicable schemes by requiring secure context, but perhaps we don't need to explicitly mention to check that in this algorithm.